spec/models/mixins/drift_state_mixin_spec.rb

```diff
@@ -0,0 +1,96 @@
+describe DriftStateMixin do
+  include Spec::Support::ArelHelper
+
+  let(:host) { FactoryGirl.create(:host) }
+
+  let(:drift_states) do
+    [
+      FactoryGirl.create(:drift_state, :resource => host, :timestamp => recent_timestamp, :data => "bogus"),
+      FactoryGirl.create(:drift_state, :resource => host, :timestamp => old_timestamp, :data => "bogus")
+    ]
+  end
+
+  let(:recent_timestamp) { 2.months.ago.change(:usec => 0) }
+  let(:old_timestamp) { 4.months.ago.change(:usec => 0) }
+
+  describe "#first_drift_state" do
+    it "uses the most recent value" do
+      drift_states
+      expect(host.last_drift_state.timestamp).to eq(recent_timestamp)
+    end
+  end
+
+  describe "#last_drift_state" do
+    it "uses the least recent value" do
+      drift_states
+      expect(host.first_drift_state.timestamp).to eq(old_timestamp)
+    end
+  end
+
+  describe "#last_drift_state_timestamp" do
+    describe "with no timestamps" do
+      before { host }
+      it "has nil with sql" do
+        expect(virtual_column_sql_value(Host, "last_drift_state_timestamp")).to be_nil
+      end
+
+      it "has nil with ruby" do
+        expect(host.last_drift_state_timestamp).to be_nil
+      end
+    end
+
+    describe "with no timestamps" do
```

Comment:
Uh, this is the same description as above.  Should this be "describing" something else?